### PR TITLE
feat: add qr code switcher to the issuer node

### PR DESCRIPTION
**Pull request by x1m3**
Ability to scan QR codes in modes: Link redirection, raw or base64 encoded.

### CODE REVIEW

File: ui/src/adapters/api/connections.ts [hunk 1]
```diff
@@ -52,16 +57,18 @@ export async function getConnection({
 export async function getConnections({
   credentials,
   env,
-  params,
+  params: { max_results, page, query },
```
Comment (amonsosanz) [inline ui/src/adapters/api/connections.ts:60]: We should apply also here what Elias mentioned in [this comment](https://github.com/0xPolygonID/issuer-node/pull/564#discussion_r1431197960).

File: ui/src/adapters/api/connections.ts [hunk 2]
```diff
@@ -70,17 +77,22 @@ export async function getConnections({
...
     return buildSuccessResponse(
-      getListParser(connectionParser)
-        .transform(({ failed, successful }) => ({
-          failed,
-          successful: successful.sort((a, b) => b.createdAt.getTime() - a.createdAt.getTime()),
+      getResourceParser(connectionParser)
+        .transform(({ items: { failed, successful }, meta }) => ({
+          items: {
+            failed,
+            successful: successful.sort((a, b) => b.createdAt.getTime() - a.createdAt.getTime()),
```
Comment (amonsosanz) [inline ui/src/adapters/api/connections.ts:93]: We should apply also here what Elias mentioned in [this comment](https://github.com/0xPolygonID/issuer-node/pull/564#discussion_r1431203737).

File: ui/src/components/connections/ConnectionsTable.tsx [hunk 3]
```diff
@@ -14,6 +14,7 @@ import {
 import { useCallback, useEffect, useState } from "react";
 import { generatePath, useNavigate, useSearchParams } from "react-router-dom";
 
+import { positiveIntegerFromStringParser } from "../../adapters/parsers";
```
Comment (amonsosanz) [inline ui/src/components/connections/ConnectionsTable.tsx:17]: We always use the absolute path on imports.

File: ui/src/components/credentials/CredentialIssuedQR.tsx [hunk 4]
```diff
@@ -46,27 +48,30 @@ export function CredentialIssuedQR() {
...
+  if (!isAsyncTaskDataAvailable(issuedQRCodes)) {
     return <LoadingResult />;
   }
 
+  const [issuedQRCodeLink, issuedQRCodeRaw] = issuedQRCodes.data;
   return (
     <CredentialQR
-      qrCode={issuedQRCode.data.qrCodeLink}
-      schemaType={issuedQRCode.data.schemaType}
+      qrCodeBase64={`iden3comm://?i_m=${btoa(issuedQRCodeRaw.qrCode)}`}
```
Comment (amonsosanz) [inline ui/src/components/credentials/CredentialIssuedQR.tsx:71]: I'd remove this prop and move this logic to the component `CredentialQR`

File: ui/src/components/credentials/IssuanceMethodForm.tsx [hunk 5]
```diff
@@ -52,11 +52,16 @@ export function IssuanceMethodForm({
 
   const fetchConnections = useCallback(
     async (signal: AbortSignal) => {
-      const response = await getConnections({ credentials: false, env, signal });
+      const response = await getConnections({
+        credentials: false,
+        env,
+        params: { max_results: 0 },
```
Comment (amonsosanz) [inline ui/src/components/credentials/IssuanceMethodForm.tsx:58]: Why this `max_results: 0`?

File: ui/src/adapters/api/connections.ts [hunk 6]
```diff
@@ -4,10 +4,15 @@ import { z } from "zod";
 import { Response, buildErrorResponse, buildSuccessResponse } from "src/adapters";
```
Comment (amonsosanz) [inline ui/src/adapters/api/connections.ts:4]: Can you please run `npm run lint:fix && npm run format` and commit the changes?

File: ui/src/adapters/api/connections.ts (hunk 2 above)
Comment (x1m3) [reply ui/src/adapters/api/connections.ts:93]: Done

File: ui/src/components/connections/ConnectionsTable.tsx (hunk 3 above)
Comment (x1m3) [reply ui/src/components/connections/ConnectionsTable.tsx:17]: Good catch. Changed

File: ui/src/components/credentials/CredentialIssuedQR.tsx (hunk 4 above)
Comment (x1m3) [reply ui/src/components/credentials/CredentialIssuedQR.tsx:71]: I agree. Done,

File: ui/src/components/credentials/IssuanceMethodForm.tsx (hunk 5 above)
Comment (x1m3) [reply ui/src/components/credentials/IssuanceMethodForm.tsx:58]: Removing it. Not needed. Probably I added it when debugging

Comment (x1m3) [reply ui/src/components/credentials/IssuanceMethodForm.tsx:58]: My intention was to download all connections from server to be shown in a  select. But, I think that this select is now an input text, so I think that this function has no sense right now. And can be a performance problem if the number of connections is too big.

Is this screen
imagen

Comment (amonsosanz) [reply ui/src/components/credentials/IssuanceMethodForm.tsx]: As [discussed in Slack](https://0xpolygon.slack.com/archives/C05AG4LR7ML/p1705679168364929?thread_ts=1705655683.487219&cid=C05AG4LR7ML) there's a bug in the API that returns `0` in `max_results`, making the response invalid, which means that this dropdown cannot be populated.

As agreed, we will set this page size (`max_results`) to `1000` and will sort the connection `DESC` so users will see the most recent `1000` connections listed, which we think covers most use cases.

File: ui/src/adapters/api/connections.ts (hunk 2 above)
Comment (amonsosanz) [reply ui/src/adapters/api/connections.ts]: mmm... I just realized we can indeed completely remove the `transform`. Would you please remove it also in `getCredentials`?

File: ui/src/components/credentials/IssuanceMethodForm.tsx (hunk 5 above)
Comment (x1m3) [reply ui/src/components/credentials/IssuanceMethodForm.tsx]: Error fixed. Now server returns 1000 rows if parameter maxRows is not present.

File: ui/src/adapters/api/connections.ts (hunk 2 above)
Comment (x1m3) [reply ui/src/adapters/api/connections.ts]: sure. Removed!

Comment (amonsosanz) [reply ui/src/adapters/api/connections.ts]: You removed the transform from `getLinks` not from `getCredentials`.

Comment (x1m3) [reply ui/src/adapters/api/connections.ts]: Fixed. Transform restored in getLinks and removed from getCredentials

**Review by amonsosanz — APPROVED** [review]

**Pull request merged**